Makefile

```diff
@@ -83,7 +83,7 @@ build-e2e:
 
 .PHONY: test-e2e
 test-e2e: images build-e2e e2e-provision ## Run end-to-end test
-	hack/test-e2e.sh || $(MAKE) e2e-clean
+	hack/test-e2e.sh || ($(MAKE) e2e-clean && false)
```

Comment:
Why `&& false`?

Reply:
Since if `hack/test-e2e.sh` fails we need to clean the libvirt resource and fail

Reply:
Why doesn't `hack/test-e2e.sh` exit with non-zero if the test fails? If it doesn't we should fix the root cause.

Reply:
Oh, I see. If it fails,we clean up but we still want to return non-zero to make.